File: 백준/Silver/11047. 동전 0/동전 0.py
N, K = map(int, input().split())
coins = [int(input()) for _ in range(N)]
sorted_coins = sorted(coins, reverse=True)
count = 0

for coin in sorted_coins:
    count += K // coin
    K %= coin
    
    if K == 0:
        break
        
print(count)

File: 백준/Silver/11047. 동전 0/동전 0.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int K = sc.nextInt();
        Integer[] coins = new Integer[N];
        for (int i = 0; i < N; i++) {
            coins[i] = sc.nextInt();
        }
        sc.close();
        Integer[] sortedCoins = coins.clone();
        Arrays.sort(sortedCoins, Collections.reverseOrder());
        int count = 0;
        for (int coin : sortedCoins) {
            count += K / coin;
            K %= coin;
            if (K == 0)
                break;
        }
        System.out.print(count);
    }
}